nodeInterface/NodeInterface.go

```diff
@@ -413,6 +417,81 @@ func (n NodeInterface) ConstructOutboxProof(c ctx, evm mech, size, leaf uint64)
 	return send, root, hashes32, nil
 }
 
+func (n NodeInterface) GasEstimateComponents(
+	c ctx,
+	evm mech,
+	from, to addr,
+	ignoreTo bool,
+	gas uint64,
+	maxFeePerGas, maxPriorityFeePerGas, value huge,
+	data []byte,
+) (uint64, uint64, uint64, uint64, error) {
+
+	node, err := arbNodeFromNodeInterfaceBackend(n.backend)
+	if err != nil {
+		return 0, 0, 0, 0, err
+	}
+
+	context := n.context
+	chainid := evm.ChainConfig().ChainID
+	backend := node.Backend.APIBackend()
+	gasCap := backend.RPCGasCap()
+	block := rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(n.header.Number.Int64()))
+	nonce := uint64(0)
+
+	args := arbitrum.TransactionArgs{
+		ChainID:              (*hexutil.Big)(chainid),
+		From:                 &from,
+		Gas:                  (*hexutil.Uint64)(&gas),
+		MaxFeePerGas:         (*hexutil.Big)(maxFeePerGas),
+		MaxPriorityFeePerGas: (*hexutil.Big)(maxPriorityFeePerGas),
+		Value:                (*hexutil.Big)(value),
+		Nonce:                (*hexutil.Uint64)(&nonce),
```

Comment:
Same for the rest of these (not including Data), they should be specified as part of the `eth_call` itself not the data of the eth_call.

Reply:
I've copied everything except `To`, since that's `nodeInterface`